File: clojure/bob/src/bob.clj
(ns bob)

(defn response-for
  "Returns the proper response form Bob."
  [statement]
  ;; This map is not really necessary, but it does seperate out data
  ;; and code, and lets me play with Clojure maps.
  (def bob-dict (hash-map :question "Sure."
                          :yell "Whoa, chill out!"
                          :blank "Fine. Be that way!"
                          :allelse "Whatever."
                          ))
  
  (cond
    (and (not (re-find #"[a-z]" statement))
         (re-find #"[A-Z]" statement))
    (bob-dict :yell)
    
    (re-find #"\?$"  statement)
    (bob-dict :question)
    
    (not (re-find #"[\S]" statement))
    (bob-dict :blank)

    :else (bob-dict :allelse)))

File: python/bob/bob.py
from typing import Any

def response(hey_bob: Any) -> str :
    """Surley responses from Bob"""
    
    match hey_bob:
        case str() as reply if not reply.strip():
            return "Fine. Be that way!"
        case str() as reply if reply.isupper() and reply.rstrip().endswith("?"):
            return "Calm down, I know what I'm doing!" 
        case str() as reply if reply.isupper():
            return "Whoa, chill out!"
        case str() as reply if reply.rstrip().endswith("?"):
            return "Sure." 
        case _:
            return "Whatever."
